packages/terra-list/src/List.jsx

```diff
@@ -39,8 +39,13 @@ const List = ({
     customProps.className,
   ]);
 
+  let roleAttr;
+  if (children.length !== 0) {
```

Comment:
This logic conflicts with how we are setting a [role in the menu component](https://github.com/cerner/terra-core/blob/8c34cdca8c56eaaeb16b26a2b780e97934f662aa/packages/terra-menu/src/_MenuContent.jsx#L312). If a custom role is set like we do in menu, we should use that first, then apply this logic.